optuna/_hypervolume/hssp.py

```diff
@@ -44,10 +44,12 @@ def _solve_hssp_on_unique_loss_vals(
     subset_size: int,
     reference_point: np.ndarray,
 ) -> np.ndarray:
-    assert not np.any(reference_point - rank_i_loss_vals <= 0)
+    if not np.isfinite(reference_point).all():
+        return rank_i_indices[:subset_size]
+    diff_of_loss_vals_and_ref_point = reference_point - rank_i_loss_vals
```

Comment:
Btw, can we revert here?

Reply:
I think the current defensive code is better for when `_solve_hssp_on_unique_loss_vals` is called from other code in the future.

Reply:
Ok, then let's keep the current code!